components/Text/Heading.tsx

```diff
@@ -1,61 +1,28 @@
-import { memo, useMemo } from 'react';
+import { memo } from 'react';
 import styled, { css } from 'styled-components';
 
 type HeadingProps = {
-	variant?: 'h1' | 'h2' | 'h3' | 'h4' | 'h5';
-	className?: string;
+	variant?: 'h1' | 'h2' | 'h3' | 'h4' | 'h5' | 'h6';
 	fontSize?: number;
+	className?: string;
 };
 
 const Heading: React.FC<HeadingProps> = memo(({ variant = 'h1', fontSize, ...props }) => {
-	const StyledHeading = useMemo(() => headingMap[variant], [variant]);
-
-	return <StyledHeading $fontSize={fontSize} {...props} />;
+	return <StyledHeading as={variant} $fontSize={fontSize} $variant={variant} {...props} />;
 });
 
-const commonStyles = css<{ $fontSize?: number }>`
-	line-height: 1.4;
-	letter-spacing: 0.2px;
-	margin: 0;
-	color: ${(props) => props.theme.colors.common.primaryWhite};
-	${(props) =>
-		props.$fontSize &&
-		css`
-			font-size: ${props.$fontSize}px;
-		`}
-`;
-
-const StyledH1 = styled.h1`
-	font-size: 64px;
-	${commonStyles}
-`;
-
-const StyledH2 = styled.h2`
-	font-size: 48px;
-	${commonStyles}
-`;
-
-const StyledH3 = styled.h3`
-	font-size: 32px;
-	${commonStyles}
-`;
-
-const StyledH4 = styled.h4`
-	font-size: 20px;
-	${commonStyles}
-`;
+const sizes = { h1: 30, h2: 26, h3: 23, h4: 21, h5: 19, h6: 16 } as const;
 
-const StyledH5 = styled.h5`
-	font-size: 16px;
-	${commonStyles}
+const StyledHeading = styled.h1<{
```

Comment:
Not sure it makes sense for everything to be based off h1 if we are passing variants like h2, h3 etc?

Reply:
The `as` prop handles this for us. See line 11.